programs/invest-in-sol/src/instructions/lend.rs

```diff
@@ -0,0 +1,179 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    token::{Token, TokenAccount, Mint, mint_to},
+    associated_token::AssociatedToken,
+};
+use crate::{state::*, error::*};
+use crate::instructions::loopscale::cpi::{strategy::*, vault::*};
+
+#[derive(Accounts)]
+pub struct LendTreasurySol<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    /// treasury PDA that holds the SOL
+    #[account(
+        mut,
+        seeds = [b"treasury"],
+        bump = treasury.bump,
```

Comment:
Currently the `Treasury` state isn't actually holding a `bump` field. The `Config` state holds the `treasury_bump`.

So we'll need to move the `treasury_bump` off of `Config` and add `bump` to `Treasury`, or pull in the `Config` account and reference `config.treaury_bump` here.